### PR TITLE
Update migration docs

### PR DESCRIPTION
**Pull request by kasperski95**

### CODE REVIEW

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 1]
```diff
@@ -5,21 +5,18 @@ sidebar_label: Migrations
 # Migrations
 
 :::warning
-This feature is actively developed. Many new additions or changes will land in future Protostar releases.
+Breaking changes can be introduced without depreciation.
```
Comment (Arcticae) [inline website/docs/tutorials/06-deploying/02-migrations/README.md:8]: i liked the old phrasing better - or we can do a bit of both. We want to tell the users that it's actively developed, and to expect breaking changes

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 2]
```diff
@@ -5,21 +5,18 @@ sidebar_label: Migrations
...
 :::
 
 
-Migrations are Cairo files that help you manage contracts on the StarkNet. They are especially useful, when your project consists of multiple contracts. Migration code is mainly written in hints. Protostar injects special functions into hints' scope, which you can use to interact with the StarkNet. These functions are similar to cheatcodes used to test contracts, hence in this document we will refer to these functions as migration cheatcodes.
-
-:::info
-Protostar aims to make migration cheatcodes a subset of [testing cheatcodes](/docs/tutorials/testing/cheatcodes) in order to allow testing migration scripts against Protostar's local StarkNet.
-:::
+Migrations are Cairo files that help you manage contracts on the StarkNet.
```
Comment (Arcticae) [inline website/docs/tutorials/06-deploying/02-migrations/README.md:12]: I am missing the big picture here (what really migrations are, where do they work etc.).

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 3]
```diff
@@ -5,21 +5,18 @@ sidebar_label: Migrations
...
+They are especially useful, when your project consists of multiple contracts or your project implements [Proxy Pattern](https://blog.openzeppelin.com/proxy-patterns/). Migration code is mainly written in hints.
```
Comment (Arcticae) [inline website/docs/tutorials/06-deploying/02-migrations/README.md:13]: Why?

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 4]
```diff
@@ -5,21 +5,18 @@ sidebar_label: Migrations
...
+Protostar injects special functions into hints' scope, which you can use to interact with the StarkNet.
```
Comment (Arcticae) [inline website/docs/tutorials/06-deploying/02-migrations/README.md:14]: For what?

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/06-deploying/02-migrations/README.md (hunk 1 above)
Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md:8]: Screen Shot 2022-10-04 at 18 06 47
https://www.youtube.com/watch?v=3duFa_5Zq6A

File: website/docs/tutorials/06-deploying/02-migrations/README.md (hunk 2 above)
Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md:12]: Done

File: website/docs/tutorials/06-deploying/02-migrations/README.md (hunk 3 above)
Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md]: Update this section

File: website/docs/tutorials/06-deploying/02-migrations/README.md (hunk 4 above)
Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md]: Updated this section

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 5]
```diff
@@ -79,6 +76,12 @@ If you build the project, Protostar will print migration logs in the command lin
   transaction_hash     0x1234...
 [INFO] Migration completed
 ```
+
+## Lack of Atomicity
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md:83]: Added a snippet

File: website/docs/tutorials/06-deploying/02-migrations/README.md (hunk 1 above)
Comment (Arcticae) [reply website/docs/tutorials/06-deploying/02-migrations/README.md]: Can't argue with that

File: website/docs/tutorials/06-deploying/02-migrations/README.md [hunk 6]
```diff
@@ -5,30 +5,30 @@ sidebar_label: Migrations
...
 
 
-Migrations are Cairo files that help you manage contracts on the StarkNet. They are especially useful, when your project consists of multiple contracts. Migration code is mainly written in hints. Protostar injects special functions into hints' scope, which you can use to interact with the StarkNet. These functions are similar to cheatcodes used to test contracts, hence in this document we will refer to these functions as migration cheatcodes.
-
-:::info
-Protostar aims to make migration cheatcodes a subset of [testing cheatcodes](/docs/tutorials/testing/cheatcodes) in order to allow testing migration scripts against Protostar's local StarkNet.
-:::
+Migrations are Cairo files that help you manage contracts on the StarkNet.
+These files are responsible for staging your deployment tasks, and they're written under the assumption that your project uses [Proxy Pattern](https://blog.openzeppelin.com/proxy-patterns/).
+As your project evolves, you'll create new migration scripts to further this evolution on the blockchain.
```
Comment (Arcticae) [inline website/docs/tutorials/06-deploying/02-migrations/README.md:14]: I don't like this phrasing. I wouldn't rely on anyone understanding usage of further as a verb

Comment (kasperski95) [reply website/docs/tutorials/06-deploying/02-migrations/README.md]: [update the blockchain sentence](https://github.com/software-mansion/protostar/pull/927/commits/92a45b5930e2659071850f5180add853b3f39a69)

**Pull request merged**